### ISSUE TITLE
[astro] Build is broken with `astro@^2.5.1`

### ISSUE THREAD

**Issue by dertieran** [solidjs/solid-start#894]
For some reason the content of `/dist/server/manifest.json` changes when upgrading from [`astro@2.5.0`](https://github.com/withastro/astro/releases/tag/astro%402.5.0) to [`astro@2.5.1`](https://github.com/withastro/astro/releases/tag/astro%402.5.1) _(or higher)_

This causes the build command to fail with

```
[build] Rearranging server assets...
 error   Cannot read properties of undefined (reading 'file')
  File:
   /node_modules/.pnpm/solid-start@0.3.0-alpha.5_@solidjs+meta@0.28.5_@solidjs+router@0.8.2_astro@2.5.1_solid-js@1.7.5_vite@4.3.8/node_modules/solid-start/astro/builder.js:19:75
  Code:
    18 |     const key = Object.keys(serverManifest).find(key => key.startsWith("_all."));
    > 19 |     process.env.START_ENTRY_STATIC = join(serverPath, serverManifest[key].file);
         |                                                                           ^
      20 |   }
      22 |   if (!resolved.solidOptions.ssr) {
```

Here is the `manifest.json` from building the bare example for comparison.

<details>
<summary><code>astro@2.5.0</code></summary>

```json
{
  "_all.3aaae4bd.mjs": {
    "css": ["_astro/handler.9a7c1663.css"],
    "file": "chunks/pages/all.3aaae4bd.mjs"
  },
  "_astro.0da788a9.mjs": {
    "file": "chunks/astro.0da788a9.mjs"
  },
  "handler.css": {
    "file": "_astro/handler.9a7c1663.css",
    "src": "handler.css"
  }
}
```

</details>

<details>
<summary><code>astro@2.5.1</code></summary>

```json
{
  "_astro.f6e3e0ef.mjs": {
    "file": "chunks/astro.f6e3e0ef.mjs"
  },
  "handler.css": {
    "file": "_astro/handler.9a7c1663.css",
    "src": "handler.css"
  },
  "node_modules/.pnpm/solid-start@0.3.0-alpha.5_@solidjs+meta@0.28.5_@solidjs+router@0.8.2_astro@2.5.1_solid-js@1.7.5_vite@4.3.8/node_modules/solid-start/astro/handler.js": {
    "css": ["_astro/handler.9a7c1663.css"],
    "file": "chunks/pages/handler.js.2c8f58eb.mjs",
    "isDynamicEntry": true,
    "src": "node_modules/.pnpm/solid-start@0.3.0-alpha.5_@solidjs+meta@0.28.5_@solidjs+router@0.8.2_astro@2.5.1_solid-js@1.7.5_vite@4.3.8/node_modules/solid-start/astro/handler.js"
  }
}
```

</details>

Other dependencies
```
dependencies:
@astrojs/node 5.1.4
@solidjs/meta 0.28.5
@solidjs/router 0.8.2
solid-js 1.7.5
solid-start 0.3.0-alpha.5

devDependencies:
@types/node 20.2.4
typescript 5.0.4
```

**Comment by ryansolid** [comment]
Thanks. I was just doing this upgrade and hit this as well. I will see what is going on.

**Comment by ryansolid** [comment]
It should be fixed with the latest. I've released 0.3.0-alpha.6.
